astropy/table/tests/test_item_access.py

```diff
@@ -251,6 +251,22 @@ def test_np_integers(self, table_data):
         t = table_data.Table(table_data.COLS)
         idxs = np.random.randint(len(t), size=2)
         item = t[idxs[1]]
+        
+    def test_np_scalar(self, table_data):
+        """
+        Select rows using numpy scalars.
+        """
+        t = table_data.Table(table_data.COLS)
+        idx = np.int64(np.random.randint(len(t)))
+        item = t[idx]
+        
+    def test_np_array_scalar(self, table_data):
+        """
+        Select rows using numpy array scalars.
+        """
+        t = table_data.Table(table_data.COLS)
+        idx = np.array(np.random.randint(len(t)))
+        item = t[idx]
```

Comment:
Not sure what you mean with *successfully typecast a numpy array scalar to an int* ?
Indexing rows should work the same way with either an integer, a Numpy integer or an array scalar, so it makes sense to run the same test (`test_row`) for all. If you are not familiar with `pytest.mark.parametrize`, the idea is to run 3 times the same test with a parameter for the different indexing options, but it's just a suggestion.